src/util/check.h

```diff
@@ -95,4 +95,14 @@ T&& inline_assertion_check(LIFETIMEBOUND T&& val, [[maybe_unused]] const char* f
         format_internal_error("Unreachable code reached (non-fatal)", \
                               __FILE__, __LINE__, __func__, PACKAGE_BUGREPORT))
 
+/**
+ * UNREACHABLE is a macro that is used to mark unreachable code. It aborts the program.
+ * This is used to mark code that is not yet implemented or is not yet reachable.
+ */
+#define UNREACHABLE()                                                                                                                                   \
+    do {                                                                                                                                                \
+        std::cerr << format_internal_error("Unreachable code reached (fatal, aborting)", __FILE__, __LINE__, __func__, PACKAGE_BUGREPORT) << std::endl; \
```

Comment:
Maybe move the implementation/formatting to the cpp file?

Reply:
Is it worth for a line of code?

Reply:
This approach requires that `libbitcoinconsensus.la` being linked against `libbitcoin_util.a`. The latter, in turn, must be converted into a Libtool archive.

Therefore, I'd prefer to keep this in the header.